Week-3/Exercise3.1/Exercise3.1.js

```diff
@@ -0,0 +1,27 @@
+function memoize(func){
+    const cache = new Map();
+    return function(...args) {
+        const key = args.toString();
+        if(cache.has(key)) {
+            return cache.get(key);
+        }
+        cache.set(key, func(...args));
+        return cache.get(key);
+    } 
+ }
+
```

Comment:
The below add function can only accepts 2 variables hence we can only add 2 numbers. But as per Guideline 4 -> The "function should be able to take n number of argument" .

Reply:
@chirag-udeshi Made the changes for this.

Reply:
Looks good.